mobile/ios/WirelessDebug/WirelessDebugger/LogStreamer.swift

```diff
@@ -0,0 +1,81 @@
+//
+//  WirelessDebugger.swift
+//  Wireless Debug iOS Test App
+//
+
+import Foundation
+
+/// Main class for the Wireless Debugger library. This class handles capturing
+/// logs and sends them to the WebSocketMessenger.
+class LogStreamer {
+    private static var logStreamer: LogStreamer?
+
+    private var webSocketMessenger: WebSocketMessenger?
+    private var timeInterval: UInt32
+    
+    private let stderrPipe = Pipe()
+    private var dispatchQueue = DispatchQueue(label: "LogStreamer", attributes: .concurrent)
+
+    /// Initialize the Wireless Debugger library.
+    ///
+    /// - Parameters:
+    ///   - hostname: root hostname to connect to
+    ///   - apiKey: the API Key provided by the web interface
+    ///   - timeInterval: the interval at which to send logs
+    private init(_ hostname: String, apiKey: String, timeInterval: Int, verbose: Bool) {
+        self.webSocketMessenger = WebSocketMessenger("ws://\(hostname)/ws",
+                                                     apiKey: apiKey,
+                                                     verbose: verbose)
+        self.timeInterval = UInt32(timeInterval)
+
+        // Create a Pipe and capture stderr.
+        dup2(self.stderrPipe.fileHandleForWriting.fileDescriptor, fileno(stderr))
+
+        // Poll for data coming in on stderr. When logs come in, enqueue them
+        // for sending to the server and re-output them on stdout.
+        self.dispatchQueue.async(execute: self.readLogs)
+
+        // Every time interval, send the logs if enough time has elapsed.
+        self.dispatchQueue.async(execute: self.sendLogs)
```

Comment:
Unless I make everything static, I can't really move anything in this function to `start`. It relies on having access to the `stderrPipe` and `dispatchQueue` variables on the instance of the `LogStreamer` class.